DC-600 Image optimization

eltesoro-koji opened the pull request:
DC-600 Add screenshot optimization recommendations for open graph

rasienko reviewed (changes requested):
DC-600 #comment This information would fit better with the other optimization recommendations, listed as bullets above the image. As a tip below the image, it seems to get a bit lost. Also, I would add some information about how to optimize the image (with URL parameters). You can add an example and/or link to more info. Let me know if you have questions.

eltesoro-koji commented:
> 
> 
> DC-600 #comment This information would fit better with the other optimization recommendations, listed as bullets above the image. As a tip below the image, it seems to get a bit lost. Also, I would add some information about how to optimize the image (with URL parameters). You can add an example and/or link to more info. Let me know if you have questions.

Moved the tip to be part of the list. Also rewrote the bullets to make them consistent.

I am reluctant to mention a specific third-party tool (as mentioned in the Jira) for several reasons, including the following:
* It implies that we officially recommend that specific tool, and, if it turns out to have malware, our reputation (by association) goes with it.
* We alienate other third-party tools who might want to partner with us in the future.

rasienko commented on src/asciidoc/docs/publish/richPreview.adoc:
```diff
@@ -18,15 +18,13 @@ You can also customize the sharing metadata by configuring an entitlement.
-* Includes customizable elements to generate unique previews of each remix.
+* The image is responsive.
+* The image looks good with a square aspect ratio.
+* The image is optimized. For example, you can use a third-party tool to reduce the image size to 400x400 pixels or load only the first frame of a gif.
```
Might want to say what you mean by "optimized" (i.e., specifically what it's optimized for). Something about loading quickly.

eltesoro-koji replied:
Fixed.

rasienko reviewed (changes requested):
There seems to be a misunderstanding. Fastly is built in to our APIs (i.e., our platform uses it to provide optimization options). I agree that you don't need to mention Fastly (since we could change providers). However, you should definitely provide examples that use our APIs.

eltesoro-koji commented:
>> There seems to be a misunderstanding. Fastly is built in to our APIs (i.e., our platform uses it to provide optimization options). I agree that you don't need to mention Fastly (since we could change providers). However, you should definitely provide examples that use our APIs.

Rachel, I did research in the Reference section for this.
1. The options for optimizing images are part of `ui.capture` under `CaptureImageOptions` (https://developer.withkoji.com/reference/core/core-frontend-ui-capture#CaptureImageOptions). I don't see the `frame` option anywhere else.
2. UI / Capture "captures user input on the frontend of your Koji." But this topic is about the image displayed when a template is shared, which means the image is displayed *before* any user has interacted with the template.

Am I missing something?

rasienko commented on src/asciidoc/docs/publish/richPreview.adoc:
```diff
@@ -18,9 +18,11 @@ You can also customize the sharing metadata by configuring an entitlement.
+* The image is responsive.
+* The image looks good with a square aspect ratio.
+* The image is optimized for faster loading.
+For example, to reduce the image size to 400x400 pixels and load only the first frame of a gif, you can add the query parameters `fit=bounds&width=400&height=400&frame=1` at the end of your image's URL: `https://images.koji-cdn.com/path/to/myimage.gif?fit=bounds&width=400&height=400&frame=1`
```
Yes, this is good! Quick things...

- Escape the link so that it's not rendered as a live link (use the `\` before the protocol).
- Consider demoing the parameters only once rather than twice. For example:
```suggestion
For example, to reduce the image size to 400x400 pixels and load only the first frame of a gif, you can add query parameters at the end of your image's URL: `\https://images.koji-cdn.com/path/to/myimage.gif?fit=bounds&width=400&height=400&frame=1`
```

rasienko reviewed (changes requested):
Quick fix and then we're good to go!
